router/config.go

```diff
@@ -880,6 +880,42 @@ func LoadConfigWithOptions(path string, loadIdentity bool) (*Config, error) {
 		return nil, err
 	}
 
+	if loadIdentity {
+		var errs []error
+		for _, c := range cfg.Link.Listeners {
+			a := c["advertise"]
+			if a != nil {
+				// should start with tls:
```

Comment:
"Should" or "can?" My naive observation has been that the `tls:` prefix is implied if omitted, and that we don't currently support any alternatives, so there's not currently any ambiguity, but there will be ambiguity in the future if we support another on-the-wire protocol, e.g., dtls.